pkg/tracegen/intake.go

```diff
@@ -26,24 +26,25 @@ import (
 	"time"
 
 	"go.elastic.co/apm/v2"
-	"go.uber.org/zap"
 )
 
-// IndexIntakeV2Trace generate a trace including a transaction, a span and an error
-func IndexIntakeV2Trace(ctx context.Context, cfg Config, logger *zap.SugaredLogger) (apm.TraceContext, error) {
-	logger.Info("sample rate set to ", cfg.SampleRate)
+// SendIntakeV2Trace generate a trace including a transaction, a span and an error
+func SendIntakeV2Trace(ctx context.Context, cfg Config) (apm.TraceContext, error) {
```

Comment:
Same here, it's preferable to have functions validate that what they need is set.